scene-lib/src/scenelib/annotations/el/LocalLocation.java

```diff
@@ -1,91 +1,217 @@
 package scenelib.annotations.el;
 
+import org.objectweb.asm.Label;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
 import java.util.Objects;
 
 /**
  * A {@link LocalLocation} holds location information for a local
- * variable: slot index, scope start, and scope length.
+ * variable: slot index, scope start, and scope end.
  */
 public final class LocalLocation {
     /**
-     * The slot index of the local variable.
+     * The start of the scopes of the element being visited.
+     * Used only for TypeReference#LOCAL_VARIABLE and TypeReference#RESOURCE_VARIABLE.
+     */
+    public final Label[] start;
+
+    /**
+     * The end of the scopes of the element being visited.
+     * Used only for TypeReference#LOCAL_VARIABLE and TypeReference#RESOURCE_VARIABLE.
      */
-    public final int index;
+    public final Label[] end;
 
     /**
-     * The start of the local variable's scope (or live range), as an offset
-     * from the beginning of the method code in bytes.
+     * The indices of the element being visited in the classfile.
+     * Used only for TypeReference#LOCAL_VARIABLE and TypeReference#RESOURCE_VARIABLE.
      */
-    public final int scopeStart;
+    public final int[] index;
 
     /**
-     * The length of the local variable's scope (or live range), in bytes.
+     * The name of the local variable being visited.
      */
-    public final int scopeLength;
+    public final String variableName;
 
-    public final String varName;
-    public final int varIndex;
+    // We now use the public ASM bytecode framework instead of our own local copy.
```

Comment:
Please remove this comment.  It's useful to you during refactoring, but will be confusing to a programmer reading the source code after the pull request is merged.